client/router.js

```diff
@@ -131,7 +130,7 @@ export default function getRouter() {
     if (to.matched.some(it => it.meta.auth) && !auth.user) {
       return next({ path: '/login', query: { redirect: to.fullPath } });
     }
-    if (!isAllowed(to)) {
+    if (!isAllowed(to, auth)) {
```

Comment:
```suggestion
     return !isAllowed(to, auth) ? next({ path: from.fullPath }) : next();
```
Also you could write it in multiline ternary for for better readability.